src/overlays/actors/ovl_En_Fg/z_en_fg.c

```diff
@@ -21,40 +47,417 @@ const ActorInit En_Fg_InitVars = {
     (ActorFunc)EnFg_Update,
     (ActorFunc)EnFg_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D280.asm")
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HIT0,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x000010AA, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON | BUMP_HOOKABLE,
+        OCELEM_ON,
+    },
+    { 8, 10, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit2 sColChkInfoInit2 = {
+    1, 0, 0, 0, MASS_IMMOVABLE,
+};
+
+static DamageTable sDamageTable = {
+    0x00, 0x11, 0x00, 0x01, 0x00, 0xE1, 0x11, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x31, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
+static EnFgAnimation sAnimations[] = {
+    { &D_06001534, 1.0f, 0, -1, 0, 0 },
+    { &D_06001534, 1.0f, 0, -1, 0, -4 },
+    { &D_060011C0, 1.0f, 0, -1, 0, -4 },
+    { &D_060007BC, 1.0f, 0, -1, 2, -4 },
+};
+
+s32 func_80A2D280(SkelAnime* skelAnime, s16 animIndex) {
+    s16 frameCount;
+    s32 ret;
+
+    ret = false;
+    if (animIndex >= 0 && animIndex < 4) {
+        ret = true;
+        frameCount = sAnimations[animIndex].frameCount;
+        if (frameCount < 0) {
+            frameCount = SkelAnime_GetFrameCount((GenericAnimationHeader*)sAnimations[animIndex].animationSeg);
```

Comment:
Personally I would prefer `&sAnimations[animIndex].animationSeg->genericHeader` instead of the cast, as the `GenericAnimationHeader` is the first element of AnimationHeader.